server/sonar-server/src/test/java/org/sonar/server/qualityprofile/ws/DeleteActionTest.java

```diff
@@ -116,89 +122,176 @@ public void delete_nominal_with_language_and_name() throws Exception {
+    dbTester.getSession().commit();
+    return profile;
   }
 
   @Test
   public void throw_UnauthorizedException_if_not_logged_in() throws Exception {
     expectedException.expect(UnauthorizedException.class);
 
-    tester.newPostRequest("api/qualityprofiles", "delete").execute();
+    tester.newPostRequest("api/qualityprofiles", "delete")
+      .setParam("organization", organization.getKey())
+      .execute();
   }
 
   @Test
   public void fail_on_missing_arguments() throws Exception {
     logInAsQProfileAdministrator();
 
     expectedException.expect(IllegalArgumentException.class);
-    expectedException.expectMessage("Both profile language and name must be set");
+    expectedException.expectMessage("If no quality profile key is specified, language and name must be set");
 
-    tester.newPostRequest("api/qualityprofiles", "delete").execute();
+    tester.newPostRequest("api/qualityprofiles", "delete")
+      .setParam("organization", organization.getKey())
+      .execute();
   }
 
   @Test
   public void fail_on_missing_language() throws Exception {
     logInAsQProfileAdministrator();
 
     expectedException.expect(IllegalArgumentException.class);
-    expectedException.expectMessage("Both profile language and name must be set");
+    expectedException.expectMessage("If no quality profile key is specified, language and name must be set");
 
-    tester.newPostRequest("api/qualityprofiles", "delete").setParam("profileName", "Polop").execute();
+    tester.newPostRequest("api/qualityprofiles", "delete")
+      .setParam("organization", organization.getKey())
+      .setParam("profileName", "Polop").execute();
   }
 
   @Test
   public void fail_on_missing_name() throws Exception {
     logInAsQProfileAdministrator();
 
     expectedException.expect(IllegalArgumentException.class);
-    expectedException.expectMessage("Both profile language and name must be set");
+    expectedException.expectMessage("If no quality profile key is specified, language and name must be set");
 
-    tester.newPostRequest("api/qualityprofiles", "delete").setParam("language", xoo1.getKey()).execute();
+    tester.newPostRequest("api/qualityprofiles", "delete")
+      .setParam("language", xoo1.getKey())
+      .setParam("organization", organization.getKey())
+      .execute();
   }
 
   @Test
   public void fail_on_too_many_arguments() throws Exception {
     logInAsQProfileAdministrator();
+    QualityProfileDto profile = insertQualityProfile();
 
     expectedException.expect(IllegalArgumentException.class);
-    expectedException.expectMessage("Either key or couple language/name must be set");
-
-    tester.newPostRequest("api/qualityprofiles", "delete").setParam("profileName", "Polop").setParam("language", xoo1.getKey()).setParam("profileKey", "polop").execute();
+    expectedException.expectMessage("When providing a quality profile key, neither of organization/language/name must be set");
```

Comment:
"neither organization nor language nor name must be set"

or "none of organization, language or name must be set"